### PR TITLE
Hide the preview if the static file is large #7431

### PR DESCRIPTION
**Pull request by vishu1993**

### CODE REVIEW

**Comment by vishu1993** [comment]
staticfile

**Comment by vishu1993** [comment]
finalnereids3

File: view/static_file_form.xml [hunk 1]
```diff
@@ -1,7 +1,14 @@
 <data>
     <xpath expr="/form/field[@name='mimetype']" position="after">
-        <label name="is_large_file"/>
+        <label name="is_large_file" />
         <field name="is_large_file"/>
         <button name="upload_large_file" string="Upload Large File"/>
     </xpath>
-</data>
\ No newline at end of file
+    <xpath expr="/form/separator[@id='sepr_preview']" position="replace">
```
Comment (PritishC) [inline view/static_file_form.xml]: Within this `<xpath>`, you will have to re-define the separator with a `states` dict which makes it invisible if `is_large_file` is `True`. Doing a replace operation alone will remove the separator completely.

**Comment by vishu1993** [comment]
updatedscreen

**Comment by PritishC** [comment]
LGTM.

If you are sure that there is no other way to present this, then assign to PP/Tarun for merge.

File: view/static_file_form.xml [hunk 2]
```diff
@@ -1,7 +1,18 @@
 <data>
     <xpath expr="/form/field[@name='mimetype']" position="after">
-        <label name="is_large_file"/>
+        <label name="is_large_file" />
```
Comment (tarunbhardwaj) [inline view/static_file_form.xml]: stray change

File: view/static_file_form.xml [hunk 3]
```diff
@@ -1,7 +1,18 @@
...
\ No newline at end of file
+    <xpath expr="/form/separator[@id='sepr_preview']" position="replace">
+    </xpath>
+    <xpath expr="//form/field[@name='file_binary']" position="replace">
+        <separator string="Preview" colspan="4" id="sepr_preview" />
+        <group id="image_preview" states="{'invisible': Bool(Eval('is_large_file'))}"  colspan="4">
+            <field name="file_binary" widget="image" colspan="4"/>
+        </group>
+        <newline/>
+        <label id="preview" string="You cant preview large files" states="{'invisible': ~Bool(Eval('is_large_file'))}" colspan="2"/>
```
Comment (tarunbhardwaj) [inline view/static_file_form.xml]: can`'`t

**Comment by vishu1993** [comment]
updatedstatic

**Pull request merged**